src/main/java/gregtech/common/covers/filter/BaseFilterContainer.java

```diff
@@ -0,0 +1,170 @@
+package gregtech.common.covers.filter;
+
+import gregtech.api.util.IDirtyNotifiable;
+
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.math.MathHelper;
+import net.minecraftforge.common.util.INBTSerializable;
+import net.minecraftforge.items.ItemStackHandler;
+
+import com.cleanroommc.modularui.api.widget.IWidget;
+import com.cleanroommc.modularui.screen.ModularPanel;
+import com.cleanroommc.modularui.value.sync.GuiSyncManager;
+import org.jetbrains.annotations.NotNull;
+
+public abstract class BaseFilterContainer<R, T extends Filter<R>> implements INBTSerializable<NBTTagCompound> {
+
+    private final IDirtyNotifiable dirtyNotifiable;
+    private int maxTransferSize = 1;
+    private T currentFilter;
+    private Runnable onFilterInstanceChange;
+    private int transferSize;
+    protected final FilterSlotHandler filterInventory;
```

Comment:
Assuming this handler is what contains the item form of the filter, I feel like this is in the wrong place. It makes more sense to me that the object containing the filter should manage the inventory which contains the filter item.

Reply:
do you mean to move filter related fields/methods into the `FilterSlotHandler`?

Reply:
I am not sure if that is what Tech meant by his comment, but moving the Filter Inventory related methods into FilterSlotHandler does make sense to me.

Reply:
mkay, i think i've done this now, since i basically just merged the roles of the handler and container together.